algorithms/Python/graphs/bfs-sequence.py

```diff
@@ -0,0 +1,75 @@
+"""
+BFS graph using Adjecency List
+-----------------------------------------------------------------------------------------
+-In Breadth First Search Sequence of any graph all childs of a parent node 
+ is visited first and the children of all these childs are stored in the QUEUE array and 
+ VISITED array.
+-Children of nodes in QUEUE are visited one by one and stored in the QUEUE and 
+ VISITED as well.
+-When all children of a node is visited that node is removed from the QUEUE.
+-These steps are repeated till all nodes in QUEUE are exhausted.
+-----------------------------------------------------------------------------------------
+-VISITED array is required to check if a node is already in BFS sequence or not.
+-QUEUE array is important for ensuring that all nodes and edges are visited.
+------------------------------------------------------------------------------------------
+The sequence of nodes printed in every recursion is the BFS-SEQUENCE
+------------------------------------------------------------------------------------------
+"""
+class Graph:
+    queue=[] #keeps order of BFS tree
+    visited=[] #keeps visited node 
+    # front and rear for accessing queue
+    front=0
+    rear=0
+    def ShowGraph(Adj_Dict): #displays graph
+        for i in Adj_Dict:
+            print(i,"->",Adj_Dict[i])
+    def Display_BFS(curr,Adj_Dict): # displays BFS sequence
+        print(curr,end=" ")
+        if curr in Adj_Dict :
+            if curr not in Graph.visited: 
+                Graph.visited.append(curr) 
+                Graph.queue.append(curr) 
+                Graph.rear+=1
+            for i in Adj_Dict[curr]: # iterate over all neighbours of curr
+                if i not in Graph.visited: 
+                    Graph.queue.append(i)
+                    Graph.visited.append(i)
+                    Graph.rear+=1            
+        Graph.queue[Graph.front]=-1 # all nodes adjecent to curr are visited
+        Graph.front+=1
+        if Graph.front==Graph.rear: # no new node to visit
+            return
+        else:
+            Graph.Display_BFS(Graph.queue[Graph.front],Adj_Dict) # go to next node
+        return
+
+#__main__
+#Dry Run
+#g is an adjecency list in form of dictionary
+g={1:[2,4],2:[4,5],4:[7,5],5:[1,3,6],6:[3,8],8:[7]} # this is directed graph
+# for undirected graph each edge has to be given twice
+# Eg:- edge from 1-2 input as {1:[2]}
+Graph.ShowGraph(g)
```

Comment:
As there is no other indication specifying otherwise, mypy is considering `showGraph ()`and `Display_BFS ()` as Instance methods and thus require them to be called via an object instance of class `Graph` and thus generating errors as follows :

```
algorithms/Python/graphs/bfs-sequence.py:53: error: Argument 1 to "ShowGraph" has incompatible type "Dict[int, List[int]]"; expected "Graph"
algorithms/Python/graphs/bfs-sequence.py:55: error: Argument 1 to "Display_BFS" has incompatible type "int"; expected "Graph"
```

currently I can think of 2 possible ways to fix it :
1) re-implementing those methods as proper Instance methods and call them via an object instance 
or
2) explicitly specify these methods as static ( using `@staticmethod` decorator )